records/widgets.py

```diff
@@ -0,0 +1,71 @@
+from bitfield.forms import BitFieldCheckboxSelectMultiple
+from bitfield.types import BitHandler
+
+from django.forms.widgets import CheckboxFieldRenderer
+from django.utils.html import conditional_escape, format_html, html_safe
+from django.utils.safestring import mark_safe
+from django.utils.encoding import (
+    force_str, force_text, python_2_unicode_compatible,
+)
+
+
+class TagRenderer(CheckboxFieldRenderer):
+    def __init__(self, name, value, attrs, choices):
+        self.name = name
+        self.value = value
+        self.attrs = attrs
+        self.choices = choices
+        self.tags_weight = self.attrs.pop('user_tags_order', [])
+
+    def render(self, name=None, value=None, attrs=None):
+        """
+
+        Outputs a <ul> for this set of ordered choice fields.
+        If an id was given to the field, it is applied to the <ul> (each
+        item in the list will get an id of `$id_$i`).
+        """
+        id_ = self.attrs.get('id')
+        output = {}
+
+        for i, choice in enumerate(sorted(self.choices)):
```

Comment:
is it a good idea to split this long method into `convert_to_html` and `sort_tags` methods and call them from method `render`?